File: HPC/Organized/archive/numint/v3c_trap/numint.c
#include <stdio.h>
#include <math.h>
#include <sys/time.h>


double gettime()
{
    struct timeval tv;
    gettimeofday(&tv, NULL); // both are output arguments. we do not need time zone -> it is NULL
    return tv.tv_sec + 0.000001 * tv.tv_usec;
}

int main( int argc, char ** argv )
{
    typedef double myf;
    #define mycos(x) cos(x)
    // cosf: float, cos: double, cosl: long double
    typedef int myi;
    myf a = 0; // float, double, long double
    myf b = 1.570796326794896619231321691639751442L;
    myi N = (myi)atoll(argv[1]);
    myf res = 0;
    myf dx = (b-a)/N;
    myf dxhalf = dx*0.5;
    myi i;

    // loop over subintervals
    double start = gettime();
    for(i=1;i<N;i++)
    {
        res += mycos( i*dx );
    }
    res *= 2.0;
    res += mycos(a) + mycos(b);
    res *= dxhalf;
    double end = gettime();

//    # print results
    printf("Results: %18.16Lf\n", (long double)res);
//    # compare with analytic result (1)
    printf("Error: %18.16Lf\n", (long double)(res - 1.0));
    printf("Time: %8.6f\n", end - start);
    return 0;
}

File: HPC/Organized/archive/numint/v2p/numint.py
#!/usr/bin/python3
import math # need for pi, cos
import sys
import time

a = 0
b = math.pi/2
N = int(sys.argv[1])
res = 0
dx = (b-a)/N
dxhalf = dx*0.5

# loop over subintervals
start = time.time()
for i in range(0,N):
# sum the spaces of the subintervals
    res += math.cos( i*dx+dxhalf )
res *= dx
end = time.time()

# print results
print(f"Results: {res}")
# compare with analytic result (1)
print(f"Error: {res - 1}")
print(f"Time: {end - start}")
